### ISSUE TITLE
Error when try to use a copiled local flutter engine engine 

### ISSUE THREAD

**Issue by rcorbellini** [flutter/flutter#130991]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

I fork the flutter engine and dosent change nothing, just for test local engine
copiled usign 

```
./flutter/tools/gn --android --android-cpu x64 --unoptimized
inja -C out/android_debug_unopt_x64
```

then i try to run my new flutter app using that engine:
```
`flutter run --local-engine=android_debug_unopt_x64 --local-engine-src-path=/home/rcorbellini/engine/engine/src`

```

```
but i got some errors:
Launching lib/main.dart on sdk gphone64 x86 64 in debug mode...
../snap/flutter/common/flutter/packages/flutter/lib/src/scheduler/binding.dart:389:13: Error: The type 'AppLifecycleState' is not exhaustively matched by the switch cases since it doesn't match 'AppLifecycleState.hidden'.
 - 'AppLifecycleState' is from 'dart:ui'.
Try adding a default case or cases that match 'AppLifecycleState.hidden'.
    switch (state) {
            ^
../snap/flutter/common/flutter/packages/flutter/lib/src/rendering/layer.dart:2308:38: Error: 'PhysicalShapeEngineLayer' isn't a type.
        oldLayer: _engineLayer as ui.PhysicalShapeEngineLayer?,
                                     ^^^^^^^^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/rendering/layer.dart:2302:29: Error: The method 'pushPhysicalShape' isn't defined for the class 'SceneBuilder'.
 - 'SceneBuilder' is from 'dart:ui'.
Try correcting the name to the name of an existing method, or defining a method named 'pushPhysicalShape'.
      engineLayer = builder.pushPhysicalShape(
                            ^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/semantics/semantics.dart:673:60: Error: Member not found: 'doNotUseWillBeDeletedWithoutWarningValuesAsList'.
      for (final SemanticsAction action in SemanticsAction.doNotUseWillBeDeletedWithoutWarningValuesAsList)
                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/semantics/semantics.dart:685:54: Error: Member not found: 'doNotUseWillBeDeletedWithoutWarningValuesAsList'.
      for (final SemanticsFlag flag in SemanticsFlag.doNotUseWillBeDeletedWithoutWarningValuesAsList)
                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/semantics/semantics.dart:2761:46: Error: Member not found: 'doNotUseWillBeDeletedWithoutWarningValuesAsList'.
    final List<String> flags = SemanticsFlag.doNotUseWillBeDeletedWithoutWarningValuesAsList.where((SemanticsFlag flag) => hasFlag(flag)).map((SemanticsFlag flag) => flag.toString().substring('SemanticsFlag.'.length)).toList();
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/semantics/binding.dart:23:9: Error: The setter 'onSemanticsAction' isn't defined for the class 'PlatformDispatcher'.
 - 'PlatformDispatcher' is from 'dart:ui'.
Try correcting the name to the name of an existing setter, or defining a setter or field named 'onSemanticsAction'.
      ..onSemanticsAction = _handleSemanticsAction
        ^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/widgets/text_selection.dart:3262:13: Error: The type 'AppLifecycleState' is not exhaustively matched by the switch cases since it doesn't match 'AppLifecycleState.hidden'.
 - 'AppLifecycleState' is from 'dart:ui'.
Try adding a default case or cases that match 'AppLifecycleState.hidden'.
    switch (state) {
            ^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.
```
i cant understand why i am getting this.

if i run without localengine parameter works fine.

### Actual results

error when using localengine

### Logs

<details>
<summary>Logs</summary>

```console

Launching lib/main.dart on sdk gphone64 x86 64 in debug mode...
../snap/flutter/common/flutter/packages/flutter/lib/src/scheduler/binding.dart:389:13: Error: The type 'AppLifecycleState' is not exhaustively matched by the switch cases since it doesn't match 'AppLifecycleState.hidden'.
 - 'AppLifecycleState' is from 'dart:ui'.
Try adding a default case or cases that match 'AppLifecycleState.hidden'.
    switch (state) {
            ^
../snap/flutter/common/flutter/packages/flutter/lib/src/rendering/layer.dart:2308:38: Error: 'PhysicalShapeEngineLayer' isn't a type.
        oldLayer: _engineLayer as ui.PhysicalShapeEngineLayer?,
                                     ^^^^^^^^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/rendering/layer.dart:2302:29: Error: The method 'pushPhysicalShape' isn't defined for the class 'SceneBuilder'.
 - 'SceneBuilder' is from 'dart:ui'.
Try correcting the name to the name of an existing method, or defining a method named 'pushPhysicalShape'.
      engineLayer = builder.pushPhysicalShape(
                            ^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/semantics/semantics.dart:673:60: Error: Member not found: 'doNotUseWillBeDeletedWithoutWarningValuesAsList'.
      for (final SemanticsAction action in SemanticsAction.doNotUseWillBeDeletedWithoutWarningValuesAsList)
                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/semantics/semantics.dart:685:54: Error: Member not found: 'doNotUseWillBeDeletedWithoutWarningValuesAsList'.
      for (final SemanticsFlag flag in SemanticsFlag.doNotUseWillBeDeletedWithoutWarningValuesAsList)
                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/semantics/semantics.dart:2761:46: Error: Member not found: 'doNotUseWillBeDeletedWithoutWarningValuesAsList'.
    final List<String> flags = SemanticsFlag.doNotUseWillBeDeletedWithoutWarningValuesAsList.where((SemanticsFlag flag) => hasFlag(flag)).map((SemanticsFlag flag) => flag.toString().substring('SemanticsFlag.'.length)).toList();
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/semantics/binding.dart:23:9: Error: The setter 'onSemanticsAction' isn't defined for the class 'PlatformDispatcher'.
 - 'PlatformDispatcher' is from 'dart:ui'.
Try correcting the name to the name of an existing setter, or defining a setter or field named 'onSemanticsAction'.
      ..onSemanticsAction = _handleSemanticsAction
        ^^^^^^^^^^^^^^^^^
../snap/flutter/common/flutter/packages/flutter/lib/src/widgets/text_selection.dart:3262:13: Error: The type 'AppLifecycleState' is not exhaustively matched by the switch cases since it doesn't match 'AppLifecycleState.hidden'.
 - 'AppLifecycleState' is from 'dart:ui'.
Try adding a default case or cases that match 'AppLifecycleState.hidden'.
    switch (state) {
            ^
Target kernel_snapshot failed: Exception
```

</details>

### Flutter Doctor output

<details>
<summary>Doctor output</summary>

```console

lutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.5, on Ubuntu 22.04.2 LTS 5.19.0-46-generic,
    locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.77.3)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!


```

</details>

**Comment by huycozy** [comment]
Hi @rcorbellini 
I think this issue is caused by you being on 3.10.5 which doesn't contain https://github.com/flutter/engine/pull/42418. The commit https://github.com/flutter/engine/commit/81839ac81a644b58c598cbf3560eb9e76aafceeb is not available on stable yet. Please retry on Flutter master channel to see if it works.

**Comment by rcorbellini** [comment]
i upgrade flutter to 3.10.6 and fork/clone again on master the engine, and make all setup but still gettin error:

```
[✓] Flutter (Channel stable, 3.10.6, on Ubuntu 22.04.2 LTS 5.19.0-46-generic,
    locale en_US.UTF-8)
    • Flutter version 3.10.6 on channel stable at
      /home/rcorbellini/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (2 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1
```

**Comment by rcorbellini** [comment]
i switch the flutter framework to beta channel and works!

**Comment by huycozy** [comment]
The commit is not available on stable channel (3.10.6) now. You can check this out: [Where's my commit?](https://github.com/flutter/flutter/wiki/Where%27s-my-Commit%3F) and follow `Finding the Framework commit that contains Engine commit X` section. Then you can see the [related framework PR](https://github.com/flutter/flutter/pull/127890) with its [commit](https://github.com/flutter/flutter/commit/d8097ff4a119f0aa174df30d6b21809ea9373dfb) does not include stable yet. 

Closing as it's fixed and will land to stable in future release.
